File: packages/dotnet/Db/Models/CustomerContact.cs
namespace Db.Models;

public class CustomerContact
{
  public string Id { get; set; } = Guid.NewGuid().ToString();
  public string CustomerId { get; set; } = null!;
  public string FirstName { get; set; } = null!;
  public string LastName { get; set; } = null!;
  public string Email { get; set; } = null!;
  public string StreetAddress { get; set; } = null!;
  public string? StreetAddress2 { get; set; }
  public string City { get; set; } = null!;
  public string StateId { get; set; } = null!;
  public string Zip { get; set; } = null!;
  public string Phone { get; set; } = null!;
  public int ImageId { get; set; }
  public bool IsActive { get; set; } = false;
  public DateTime DateCreated { get; set; } = DateTime.UtcNow;
  public DateTime? DateUpdated { get; set; }

  // Navigation properties
  public Customer Customer { get; set; } = null!;
  public State State { get; set; } = null!;
}

File: apps/web/src/lib/models/customer-contact.ts
export type CustomerContactModel = {
  id: string;
  customerId: string;
  firstName: string;
  lastName: string;
  email: string;
  streetAddress: string;
  streetAddress2: string | null;
  city: string;
  stateId: string;
  zip: string;
  phone: string;
  imageId: number;
  isActive: boolean;
  dateCreated: string;
  dateUpdated: string | null;
};

export type CustomerContactCreateModel = Pick<
  CustomerContactModel,
  | 'customerId'
  | 'firstName'
  | 'lastName'
  | 'email'
  | 'streetAddress'
  | 'streetAddress2'
  | 'city'
  | 'stateId'
  | 'zip'
  | 'phone'
  | 'imageId'
  | 'isActive'
>;

export type CustomerContactUpdateModel = Pick<
  CustomerContactModel,
  | 'customerId'
  | 'firstName'
  | 'lastName'
  | 'email'
  | 'streetAddress'
  | 'streetAddress2'
  | 'city'
  | 'stateId'
  | 'zip'
  | 'phone'
  | 'imageId'
  | 'isActive'
>;

export type CustomerContactResponseModel = Pick<
  CustomerContactModel,
  | 'id'
  | 'customerId'
  | 'firstName'
  | 'lastName'
  | 'email'
  | 'streetAddress'
  | 'streetAddress2'
  | 'city'
  | 'stateId'
  | 'zip'
  | 'phone'
  | 'imageId'
  | 'isActive'
  | 'dateCreated'
  | 'dateUpdated'
>;
